.github/workflows/try.yml

```
@@ -1,203 +1,120 @@
 name: Try
 
 on:
-  pull_request_target:
-    types: [labeled]
   push:
-    branches: ["try", "try-linux", "try-mac", "try-wpt-mac", "try-wpt-mac-2020", "try-wpt", "try-wpt-2020", "try-windows"]
+    branches: ["try"]
+  workflow_dispatch:
+    inputs:
+      profile:
+        required: false
+        default: "release"
+        type: choice
+        options: ["release", "debug", "production"]
+      wpt-tests-to-run:
+        default: ""
+        required: false
+        type: string
+      wpt-layout:
+        required: false
+        type: choice
+        options: ["none", "2013", "2020", "all"]
+      unit-tests:
+        required: false
+        type: boolean
 
 jobs:
-  parse-comment:
-    name: Trigger Try
-    runs-on: ubuntu-latest
-    concurrency:
-      group: try-${{ github.event.number }}
+  decision:
+    name: Generate matrix
+    runs-on: ubuntu-20.04
     outputs:
       configuration: ${{ steps.configuration.outputs.result }}
     steps:
-      - uses: actions/github-script@v6
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+          sparse-checkout: |
+            python/servo/try_parser.py
+          sparse-checkout-cone-mode: false
+      - name: Generate full
+        id: full
+        run: |
+          out=$(python ./python/servo/try_parser.py)
+          echo "config<<EOF"$'\n'"$out"$'\n'EOF >> $GITHUB_OUTPUT
```

It's a little unfortunate that this is necessary. How long does this step take? Do you have an example of a run where this takes place that I can look at?

```suggestion
      - name: Get Full Configuration
        id: full_config
        run: |
          {
               echo 'config<<EOF'
               python ./python/servo/try_parser.py
               echo EOF
           } >> $GITHUB_OUTPUT
```
1. Use title-case consistent in step name.
2. Use a [more readable syntax](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-of-a-multiline-string) for the multi-line string.


Does this depend on the type of quotes that `try_parser.py` produces?

---

Times are negligible: https://github.com/sagudev/servo/actions/runs/7653133561/job/20854351846 and the only reason for this is so we do not need to hardcode full config in script.

---

There where some problems with quotes, that's why I made as is (possibly due to how GITHUB_OUTPUT is interpreted by runners). I can try your suggestion.

---

It works.